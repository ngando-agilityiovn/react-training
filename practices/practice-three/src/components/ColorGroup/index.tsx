import { Box, BoxProps, chakra, useRadio } from '@chakra-ui/react';

interface CustomRadioProps extends BoxProps {
  color: string;
}

const ColorGroup = (props: CustomRadioProps) => {
  const { color, ...radioProps } = props;
  const { state, getInputProps, getRadioProps, htmlProps, getLabelProps } = useRadio(radioProps);

  return (
    <chakra.label {...htmlProps} cursor="pointer">
      <input {...getInputProps({})} hidden />
      <Box w="16" h="16" {...getRadioProps()} color={color} m="34px 12px 0 40px">
        <Box
          _hover={{ background: color }}
          aria-label=""
          bg={color}
          w={16}
          h={16}
          rounded="100%"
          {...getLabelProps()}
          content='""'
          top="50%"
          left="50%"
          transform="translate(-50%, -50%)"
          width="120%"
          height="120%"
          borderRadius="100%"
          _after={{
            position: 'absolute',
            content: '""',
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)',
            width: '120%',
            height: '120%',
            bg: 'transparent',
            borderRadius: '100%',
            border: `4px solid  ${state.isChecked ? color : 'transparent'}`
          }}
          zIndex={-1}
        />
      </Box>
    </chakra.label>
  );
};

export default ColorGroup;
