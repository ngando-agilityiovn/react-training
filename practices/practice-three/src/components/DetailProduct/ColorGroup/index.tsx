import { Box, Flex, IconButton, Text } from '@chakra-ui/react'

interface IColorItem {
  color: string
}

interface IRadioGroup {
  data: IColorItem[]
}

const ColorGroup = ({ data }: IRadioGroup) => {
  return (
    <>
      <Text color="backgroundWarning" fontWeight="medium" mb="14px">
        Choose a Color
      </Text>
      <Flex gap="12px">
        {data.map(({ color }) => (
          <Box color={color}>
            <IconButton
              aria-label=""
              bg={color}
              w={30}
              h={30}
              rounded="full"
              position="relative"
              _after={{
                position: 'absolute',
                content: '""',
                top: '50%',
                left: '50%',
                transform: 'translate(-50%, -50%)',
                width: '120%',
                height: '120%',
                bg: 'transparent',
                borderRadius: '100%',
                border: `4px solid  ${color}`,
                zIndex: -1,
              }}
            />
          </Box>
        ))}
      </Flex>
    </>
  )
}

export default ColorGroup
