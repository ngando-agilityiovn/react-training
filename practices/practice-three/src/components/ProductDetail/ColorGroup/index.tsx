import { Box, Flex, IconButton, Text } from '@chakra-ui/react'
import { useState } from 'react'

interface IColorItem {
  color: string
}

interface IRadioGroup {
  data: IColorItem[]
}

const ColorGroup = ({ data }: IRadioGroup) => {
  const [isSelected, setIsSelected] = useState(0)
  return (
    <>
      <Text color="backgroundWarning" fontWeight="medium" mb="14px">
        Choose a Color
      </Text>
      <Flex gap="6">
        {data?.map(({ color }, index) => {
          return (
            <Box color={color} key={index}>
              <IconButton
                _hover={{ background: color }}
                aria-label=""
                bg={color}
                w={16}
                h={16}
                rounded="100%"
                position="relative"
                onClick={() => setIsSelected(index)}
                // TODO: Implement handle after
                _after={{
                  position: 'absolute',
                  content: '""',
                  top: '50%',
                  left: '50%',
                  transform: 'translate(-50%, -50%)',
                  width: '120%',
                  height: '120%',
                  bg: 'transparent',
                  borderRadius: '100%',
                  border: `4px solid  ${
                    isSelected === index
                      ? data[isSelected].color
                      : 'transparent'
                  }`,
                  zIndex: -1
                }}
              />
            </Box>
          )
        })}
      </Flex>
    </>
  )
}

export default ColorGroup
