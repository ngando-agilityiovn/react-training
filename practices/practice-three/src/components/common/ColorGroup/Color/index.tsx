import { Box, BoxProps, chakra, useRadio } from '@chakra-ui/react';
import { memo } from 'react';

interface CustomRadioProps extends BoxProps {
  color: string;
}

const Color = (props: CustomRadioProps) => {
  const { color, ...radioProps } = props;
  const { state, getInputProps, getRadioProps, htmlProps, getLabelProps } =
    useRadio(radioProps);

  return (
    <chakra.label aria-label={`radio-${color}`} {...htmlProps} cursor="pointer">
      <input {...getInputProps({})} />
      <Box
        w="16"
        h="16"
        {...getRadioProps()}
        color={color}
        m="34px 12px 0 40px"
      >
        <Box
          _hover={{ background: color }}
          bg={color}
          w={16}
          h={16}
          rounded="100%"
          {...getLabelProps()}
          content='""'
          top="50%"
          left="50%"
          transform="translate(-50%, -50%)"
          width="120%"
          height="120%"
          _after={{
            position: 'absolute',
            content: '""',
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)',
            width: '120%',
            height: '120%',
            bg: 'transparent',
            borderRadius: '100%',
            border: `4px solid ${state.isChecked ? color : 'transparent'}`
          }}
          zIndex={-1}
        />
      </Box>
    </chakra.label>
  );
};

export default memo(Color);
