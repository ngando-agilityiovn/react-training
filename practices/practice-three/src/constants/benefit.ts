export const BENEFIT_DATA = [
  {
    title:
      'Durable leather is easily cleanable so you can keep your look fresh.',
  },
  {
    title: 'Toe piece with star pattern adds durability.',
  },
]
