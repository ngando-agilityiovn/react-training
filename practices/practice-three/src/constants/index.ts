export * from './navbar'
export * from './filter'
