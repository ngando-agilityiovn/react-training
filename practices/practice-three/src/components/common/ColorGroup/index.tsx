import { HStack, Stack, Text, useRadioGroup } from '@chakra-ui/react'
import Color from './Color'

interface IColorGroup {
  colors?: string[]
}

const ColorGroup = ({ colors }: IColorGroup) => {
  const { getRadioProps, getRootProps } = useRadioGroup({
    defaultValue: '#3498db'
  })

  return (
    <Stack {...getRootProps()} borderBottom="1px solid gainsboro">
      <Text color="backgroundWarning" fontWeight="medium" mb="14px">
        Choose a Color
      </Text>
      <HStack>
        {colors?.map((item) => {
          return (
            <Color
              key={item}
              color={item}
              {...getRadioProps({ value: item })}
            />
          )
        })}
      </HStack>
    </Stack>
  )
}

export default ColorGroup
