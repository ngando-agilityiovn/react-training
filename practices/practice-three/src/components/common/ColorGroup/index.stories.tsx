import { Meta, StoryObj } from '@storybook/react'
import { BrowserRouter } from 'react-router-dom'

// Constants
import { DATA_COLOR } from '@/constants'

// Components
import ColorGroup from '.'

export default {
  title: 'Components/ColorGroup',
  component: ColorGroup,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    )
  ]
} as Meta

type Story = StoryObj<typeof ColorGroup>

export const Default: Story = {
  args: { colors: DATA_COLOR }
}
