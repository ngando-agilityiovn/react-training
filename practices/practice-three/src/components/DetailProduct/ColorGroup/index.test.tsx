import { render } from '@testing-library/react'

// Constants
import { DATA_COLOR } from '@/constants'

// Components
import ColorGroup from '.'

describe('ColorGroup component', () => {
  it('Render correcty', () => {
    const container = render(<ColorGroup dataColor={DATA_COLOR} />)
    expect(container).toMatchSnapshot()
  })
})
