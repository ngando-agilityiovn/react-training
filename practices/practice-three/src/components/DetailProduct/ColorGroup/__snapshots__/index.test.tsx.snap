// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorGroup component Render correcty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="chakra-text css-1f7e0rw"
      >
        Choose a Color
      </p>
      <div
        class="css-dtg2nl"
      >
        <div
          class="css-rk01vg"
        >
          <button
            aria-label=""
            class="chakra-button css-n7i3qc"
            type="button"
          />
        </div>
        <div
          class="css-15dtwlt"
        >
          <button
            aria-label=""
            class="chakra-button css-15o6u5l"
            type="button"
          />
        </div>
        <div
          class="css-1nknmqn"
        >
          <button
            aria-label=""
            class="chakra-button css-gpon6y"
            type="button"
          />
        </div>
        <div
          class="css-zp9sb0"
        >
          <button
            aria-label=""
            class="chakra-button css-1ojbwa3"
            type="button"
          />
        </div>
        <div
          class="css-18x00dg"
        >
          <button
            aria-label=""
            class="chakra-button css-19ln54c"
            type="button"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="chakra-text css-1f7e0rw"
    >
      Choose a Color
    </p>
    <div
      class="css-dtg2nl"
    >
      <div
        class="css-rk01vg"
      >
        <button
          aria-label=""
          class="chakra-button css-n7i3qc"
          type="button"
        />
      </div>
      <div
        class="css-15dtwlt"
      >
        <button
          aria-label=""
          class="chakra-button css-15o6u5l"
          type="button"
        />
      </div>
      <div
        class="css-1nknmqn"
      >
        <button
          aria-label=""
          class="chakra-button css-gpon6y"
          type="button"
        />
      </div>
      <div
        class="css-zp9sb0"
      >
        <button
          aria-label=""
          class="chakra-button css-1ojbwa3"
          type="button"
        />
      </div>
      <div
        class="css-18x00dg"
      >
        <button
          aria-label=""
          class="chakra-button css-19ln54c"
          type="button"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
